common/djangoapps/track/backends/oli.py

```diff
@@ -0,0 +1,115 @@
+"""OLI analytics service event tracker backend."""
+
+from __future__ import absolute_import
+
+import json
+import logging
+
+from django.contrib.auth.models import User
+
+from student.models import anonymous_id_for_user
+from track.backends import BaseBackend
+
+log = logging.getLogger(__name__)
+
+from requests_oauthlib import OAuth1Session
+
+class OLIAnalyticsBackend(BaseBackend):
+    """
+    Transmit events to the OLI analytics service
+    """
+    def __init__(self, **kwargs):
+        super(OLIAnalyticsBackend, self).__init__(**kwargs)
+
+        # the oli analytics service endpoint will be at
+        # "{}{}".format(oli_analytics_service_url, path)
+        self.oli_analytics_service_url = kwargs.get('oli_analytics_service_url', '')
+        self.path = kwargs.get('path', '')
+
+        # shared key for OLI analytics service
+        self.key = kwargs.get('key', '')
+
+        # shared secret for OLI analytics service
+        self.secret = kwargs.get('secret', '')
+
+        self.oauth = OAuth1Session(self.key, client_secret=self.secret)
+
+        # timeout for synchronous PUT attempt
+        self.timeout_in_secs = kwargs.get('timeout_in_secs', 1.0)
+
+        # only courses with id in this set will have their data sent
+        self.course_id_set = set(kwargs.get('course_ids', []))
+
+    def send(self, event):
+        """
+        Forward the event to the OLI analytics server
+        Exact API here: https://docs.google.com/document/d/1ZB-qwP0bV7ko_xJdJNX1PYKvTyYd4I8CBltfac4dlfw/edit?pli=1#
+        OAuth 1 with nonce and body signing
+        """
+        if not self.oli_analytics_service_url or not self.secret or not self.key:
+            return
+
+        # Only currently passing problem_check events, which are CAPA only
+        if event.get('event_type', '') != 'problem_check':
+            return
+
+        if event.get('event_source', '') != 'server':
+            return
+
+        context = event.get('context')
+        if not context:
+            return
+
+        course_id = context.get('course_id')
+        if not course_id or course_id not in self.course_id_set:
+            return
+
+        user_id = context.get('user_id')
+        if not user_id:
+            return
+
+        event_data = event.get('event', {})
```

Comment:
what is the purpose of setting {} as the default argument? if i'm understanding this correctly, if the key is not present it would just return in the next line anyways right?